src/t_hash.c

```diff
@@ -3117,14 +3150,38 @@ static void hexpireGenericCommand(client *c, const char *cmd, long long basetime
     hashTypeSetExInit(keyArg, hashObj, c, c->db, cmd, expireSetCond, &exCtx);
     addReplyArrayLen(c, numFields);
 
-    for (int i = 0 ; i < numFields ; i++) {
-        sds field = c->argv[numFieldsAt+i+1]->ptr;
+    fieldAt = numFieldsAt + 1;
+    while (fieldAt < c->argc) {
+        sds field = c->argv[fieldAt]->ptr;
         SetExRes res = hashTypeSetEx(hashObj, field, expire, &exCtx);
+
+        if (unlikely(res != HSETEX_OK)) {
+            /* If the field was not set, prevent field propagation */
+            rewriteClientCommandArgument(c, fieldAt, NULL);
+            fieldsNotSet = 1;
+        } else {
+            ++fieldAt;
+        }
+
         addReplyLongLong(c,res);
     }
+
     hashTypeSetExDone(&exCtx);
 
-    /* rewrite command for the replica sake */
+    nFieldsOK = c->argc - numFieldsAt - 1;
+
+    /* If all numFields were dropped, prevent command propagation */
+    if (!nFieldsOK) {
```

Comment:
Not sure I understand this. If all fields are deleted, we still need to propagate it right? Are we doing it here?

Reply:
Modified the comment and the code. Hope it is more clear:
```
    /* Avoid propagating command if not even one field was updated (Either because 
     * the time is in the past, and corresponding HDELs were sent, or conditions 
     * not met) then it is useless and invalid to propagate command with no fields */
    if (exCtx.fieldUpdated == 0) {
        preventCommandPropagation(c);
        return;
    }
```